iodata/formats/wfx.py

```diff
@@ -327,17 +327,34 @@ def load_one(lit: LineIterator) -> dict:
     }
 
 
+def prepare_dump(data: IOData):
+    """Check the compatibility of the IOData object with the WFX format.
+
+    Parameters
+    ----------
+    data
+        The IOData instance to be checked.
+    """
+    if data.mo is None:
+        raise PrepareDumpError("The WFX format requires molecular orbitals.")
+    if data.obasis is None:
+        raise PrepareDumpError("The WFX format requires an orbital basis set.")
+    if data.mo.kind == "generalized":
+        raise PrepareDumpError("Cannot write WFX file with generalized orbitals.")
+    if data.mo.occs_aminusb is not None:
+        raise PrepareDumpError("Cannot write WFX file when mo.occs_aminusb is set.")
+    for shell in data.obasis.shells:
+        if any(kind != "c" for kind in shell.kinds):
+            raise PrepareDumpError("The WFX only supports Cartesian MolecularBasis.")
```

Comment:
When we try to dump a `wfx` with spherical, should we convert spherical -> Cartesian and then dump. That's what I think most codes that dump `wfx` do internally.

Reply:
I agree. That's part of the plan in #191. I'm aiming to fix this in one of the following PRs. This exception is a not meant to stay like this.